src/openforms/submissions/models/submission.py

```diff
@@ -339,6 +339,20 @@ class Meta:
                 ),
                 name="registration_status_consistency_check",
             ),
+            models.CheckConstraint(
+                check=(
+                    models.Q(finalised_registration_backend_key="")
+                    | (
+                        ~models.Q(finalised_registration_backend_key="")
+                        & models.Q(completed_on__isnull=False)
```

Comment:
```suggestion
                        & models.Q(completed_on__isnull=True)
```

Or am I missing something?

Reply:
two cases:

1. either there is no registration backend key set at all - the value of `completed_on` doesn't matter
2. there is a non-empty registration backend key, in that case `completed_on` must be set (otherwise the submission is not completed) so it must not be `null` -> `completed_on__isnull=False`

Reply:
another way to read this is `completed_on__isnotnull`